firmware/arduino/chessboard/uart.ino

```diff
@@ -23,70 +52,79 @@ void serialEvent2()
         }
 
         // Check if the buffer is full, process input
-        if (byteNum  ==  5)
+        if (byteNum  ==  6)
         {
             // Reset buffer position
             byteNum = -1;
 
             // Process input
             // Returns true for valid input and false for invalid input. Calls movement function
-            parseMessageFromPi(buffer);
+            currentState = EXECUTING;
+            if (parseMessageFromPi(buffer))
```

Comment:
Should call `sendMessageToPi` to say that Arduino is executing before starting the movement function, since movement is blocking and it won't be able to tell the Pi that it's currently doing stuff once that whole thing starts. To that end, I think it could be cleaner if you are able to separate the parsing of the pi message from calling the movement function. I'll leave it up to you to decide if that's something you want to do, but this would allow getting the type of move/start and end points, then sending a message to the pi saying it's going to execute (or error if points are invalid), then calling the appropriate movement function. Some pseudocode:
```
if (parseMessageFromPi(buffer)) {
    sendMessageToPi(EXECUTING);
    // Here you would call appropriate move based on some global
    // variables you set in parseMessageFromPi
    doMove();  // blocks until move is finished, then sets status
} else {
    currentState = ERROR;
}
sendMessageToPi(currentState);  // can be either IDLE or ERROR
```